packages/compass-schema/src/modules/schema-analysis.spec.js

```diff
@@ -169,5 +195,46 @@ describe('schema-analyis', function () {
       );
       expect((await getResultPromise).code).to.equal(1000);
     });
+
+    it.only('returns the schema for a complex example', async function () {
+       const dataService = {
+        sample: () =>
+          Promise.resolve([
+            { foo: 1 },
+            { foo: [1, 2, 3] },
+            { foo: { bar: 1 } },
+            { foo: { bar: { baz: 1 } } },
+
+            { array: [{ monkey: 1}, { banana: 1 }] }
+          ]),
+      };
+
+      const abortController = new AbortController();
+      const abortSignal = abortController.signal;
+
+      const schema = await analyzeSchema(
+        dataService,
+        abortSignal,
+        'db.coll',
+        {},
+        {}
+      );
+
+      console.dir(schema, { depth: Infinity });
+
+      const paths = schemaToPaths(schema.fields);
+      console.dir(paths, { depth: Infinity });
+
+      /*
+        [
```

Comment:
One potential problem seems to be that we lost the field order so we can't loop through these when building the json doc or csv row - we'd have to loop through the document that came from the database or the csv fields in teh order as determined when we write the header row.